Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -55,10 +67,93 @@ extension OfflineMapAreasView {
             }
         }
         
-        /// Requests authorization to show notifications.
-        nonisolated func requestUserNotificationAuthorization() async {
-            _ = try? await UNUserNotificationCenter.current()
-                .requestAuthorization(options: [.alert, .sound])
+        /// Makes offline preplanned map models with information from the downloaded mobile map
+        /// packages for the online map.
+        func makeOfflinePreplannedMapModels() async {
+            guard let portalItemID else { return }
+            
+            let preplannedDirectory = FileManager.default.preplannedDirectory(forPortalItemID: portalItemID)
+            
+            guard let mapAreaIDs = try? FileManager.default.contentsOfDirectory(atPath: preplannedDirectory.path()) else { return }
+            
+            var mapAreas: [OfflinePreplannedMapArea] = []
+            
+            for mapAreaID in mapAreaIDs {
+                if let preplannedMapAreaID = PortalItem.ID(mapAreaID),
+                   let mapArea = await createMapArea(
+                    for: portalItemID,
+                    preplannedMapAreaID: preplannedMapAreaID
+                   ) {
+                    mapAreas.append(mapArea)
+                }
+            }
+            
+            offlinePreplannedMapModels = mapAreas.map { mapArea in
+                PreplannedMapModel(
+                    offlineMapTask: offlineMapTask,
+                    mapArea: mapArea,
+                    portalItemID: portalItemID,
+                    preplannedMapAreaID: mapArea.id!
+                )
+            }
+            .sorted(using: KeyPathComparator(\.preplannedMapArea.title))
+            .filter(\.status.isDownloaded)
+        }
+        
+        /// Creates a preplanned map area using a given portal item and map area ID to search for a corresponding
+        /// downloaded mobile map package. If the mobile map package is not found then `nil` is returned.
+        /// - Parameters:
+        ///   - portalItemID: The portal item ID.
+        ///   - preplannedMapAreaID: The preplanned map area ID.
+        /// - Returns: The preplanned map area.
+        private func createMapArea(
+            for portalItemID: PortalItem.ID,
+            preplannedMapAreaID: PortalItem.ID
+        ) async -> OfflinePreplannedMapArea? {
+            let fileURL = FileManager.default.preplannedDirectory(
+                forPortalItemID: portalItemID,
+                preplannedMapAreaID: preplannedMapAreaID
+            )
+            guard FileManager.default.fileExists(atPath: fileURL.path()) else { return nil }
```

Comment:
Might be an oddball solution, but you can use "fileURL.[checkResourceIsReachable()](https://developer.apple.com/documentation/foundation/url/1779667-checkresourceisreachable)" here instead. It is an Apple [helper method](https://github.com/apple/swift-corelibs-foundation/blob/main/Sources/Foundation/NSURL.swift#L561) that wraps the `atPath` API.

No need to change, up to you.

Reply:
Decided to leave as-is.